geostructures/collections.py

```diff
@@ -786,3 +786,34 @@ def filter_by_time(self, start_time: time, end_time: time) -> 'Track':
                 or shape.start.time() <= start_time <= end_time <= shape.end.time()
             ]
         )
+
+    def filter_impossible_journeys(self, max_speed: float) -> 'Track':
+        """
+        Filters out impossible journeys in the track based on speed thresholds.
+
+        Args:
+            max_speed (float): The maximum allowable speed (meters per second).
+
+        Returns:
+            Track: A new Track instance with only valid geoshapes, removing impossible journeys.
+        """
+        sz = len(self.geoshapes)
+        coords = [pt.centroid for pt in self.geoshapes]
+        times = [pt.dt.start for pt in self.geoshapes]  # Assuming time info is stored in pt.dt
+        i = 0
+        valid_geoshapes = [self.geoshapes[i]]  # Keep the first point as valid
+
+        for j in range(1, sz):
```

Comment:
This is the only place `sz` is used - should just be `for j in range(1, len(self)):`